cloudsim-plus/src/main/java/org/cloudbus/cloudsim/power/PowerMeter.java

```diff
@@ -0,0 +1,133 @@
+        super(simulation);
+        this.powerAwareEntitiesFn = powerAwareEntitiesFn;
+        this.startTime = 0;
+    }
+
+    /**
+     * Initializes a {@link PowerMeter} with a list of {@link PowerAware} entities.
+     */
+    public PowerMeter(Simulation simulation, List<? extends PowerAware<? extends PowerModel>> powerAwareEntities) {
+        this(simulation, () -> powerAwareEntities);
+    }
+
+    /**
+     * Initializes a {@link PowerMeter} with a single {@link PowerAware} entity.
+     */
+    public PowerMeter(Simulation simulation, PowerAware<? extends PowerModel> powerAwareEntity) {
+        this(simulation, new ArrayList<>(Arrays.asList(powerAwareEntity)));
+    }
+
+    @Override
+    protected void startEntity() {
+        schedule(this, startTime, POWER_MEASUREMENT);
+    }
+
+    @Override
+    public void processEvent(SimEvent evt) {
+        switch (evt.getTag()) {
+            case POWER_MEASUREMENT:
+                List<? extends PowerAware<? extends PowerModel>> powerAwareEntities;
+                try {
+                    powerAwareEntities = powerAwareEntitiesFn.call();
+                    powerMeasurements.add(powerAwareEntities.stream()
+                        .map(PowerAware::getPowerModel)
+                        .map(PowerModel::getPowerMeasurement)
+                        .reduce(PowerMeasurement::add)
+                        .orElse(new PowerMeasurement()));
+                } catch (Exception e) {
+                    // call() may raise, in this case we append an empty measurement
+                    e.printStackTrace();
+                    powerMeasurements.add(new PowerMeasurement());
+                }
+
+                schedule(this, measurementInterval, POWER_MEASUREMENT);
+                break;
+            case CloudSimTags.END_OF_SIMULATION:
+                this.shutdownEntity();
+                break;
+            default:
+                throw new RuntimeException("Unknown Event: " + evt);
+        }
+    }
+
+    @Override
+    public PowerMeter setName(String name) {
+        super.setName(name);
+        return this;
+    }
+
+    /**
+     * Gets the list of all measurements collected up to now.
+     * @return
+     */
+    public List<PowerMeasurement> getPowerMeasurements() {
+        return powerMeasurements;
+    }
+
+    /**
+     * Gets the time interval to collect power measurements.
+     * @return
+     */
+    public double getMeasurementInterval() {
+        return measurementInterval;
+    }
+
+    /**
+     * Sets the time interval to collect power measurements.
+     * @param measurementInterval the value to set
+     * @return
+     */
+    public void setMeasurementInterval(double measurementInterval) {
+        this.measurementInterval = measurementInterval;
+    }
+
+    /**
+     * Gets the time the power meter started to collect measurements.
+     * @return
+     */
+    public double getStartTime() {
```

Comment:
Usually, to indicate that startTime is not set yet, its default value is defined as -1.
This is an approach used for Cloudlet for instance.

Reply:
As far as I understand the code `startTime` is only used to delay the first measurement, right? It is only used in
```
startEntity() {
    schedule(this, startTime, POWER_MEASUREMENT);
}
```

Not sure why I included that, I do not really see the purpose and the name is also a little misleading (because it's not the absolute simulation time but the relative delay). I'd suggest removing this field altogether to keep things simple, it's also not used in LEAF. Do you agree?

Reply:
> As far as I understand the code `startTime` is only used to delay the first measurement, right? It is only used in
> 
> ```
> startEntity() {
>     schedule(this, startTime, POWER_MEASUREMENT);
> }
> ```

Yeah, it is. I missed that. I just updated the documentation to make it clearer.
> 
> Not sure why I included that, I do not really see the purpose and the name is also a little misleading (because it's not the absolute simulation time but the relative delay). I'd suggest removing this field altogether to keep things simple, it's also not used in LEAF. Do you agree?